File: JavaScript/1290-convert-bin-num-in-linked-list-to-dec.js
/**
 ** Definition for singly-linked list.
 ** function ListNode(val, next) {
 **     this.val = (val===undefined ? 0 : val)
 **     this.next = (next===undefined ? null : next)
 ** }
 **/
/**
 ** @param {ListNode} head
 ** @return {number}
 **/

var getDecialValue = function (head) {
    // let num = "" + head.val;
    let num = head.val;

    while (head.next) {
        // num += head.next.val;
        // num = num * 2 + head.next.val;
        num = num << 1 | head.next.val;
        head = head.next;
    }
    // return parseInt(num,2);
    return num;
}

File: Java/1290-convert-bin-num-in-linked-list-to-dec.java
/**
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode () {}
 * ListNode (int val) { this.val = val; }
 * ListNode (int val, ListNode next) { this.val = val; this.next = next }
 * }
 **/

class Solution {
    public int getDecimalValue(ListNode head) {
        // String num = "" + head.val;
        int num = head.val;
        while (head.next != null){
            // num += head.next.val;
            // num = num * 2 + head.next.val;
            num = num << 1 | head.next.val;
            head = head.next;
        }
        // return Integer.parseInt(num,2);
        return num;
    }
}
